packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx

```diff
@@ -2,156 +2,107 @@ import React, { FC, useState } from 'react';
 
-              // Returning true here triggers the slide animation
-              return true;
-            } catch (e) {
-              //
-            }
-          }}
-          disabled={mode !== ModalMode.Update}
-        />
-      }
-      okButton={
-        <DialogButton
-          variant="ok"
-          onClick={async () => {
-            try {
-              await save(draftLines);
-              onClose();
-            } catch (e) {
-              error(t('error.cant-save'))();
-            }
-          }}
-        />
-      }
-      height={600}
-      width={1024}
+  const onOk = async () => {
+    try {
+      await save(draftLines);
+      onClose();
+    } catch (e) {
+      error(t('error.cant-save'))();
+    }
+  };
+
+  return (
+    <StocktakeLineEditModal
+      onNext={onNext}
+      onOk={onOk}
+      onCancel={onClose}
+      mode={mode}
+      isOpen={isOpen}
     >
-      {!isLoading ? (
-        <>
-          <StocktakeLineEditForm
-            item={currentItem}
-            onChangeItem={setCurrentItem}
-            mode={mode}
-          />
-          <Divider margin={5} />
-          {currentItem ? (
-            <TabContext value={currentTab}>
-              <Box flex={1} display="flex" justifyContent="space-between">
-                <Box flex={1} />
-                <Box flex={1}>
-                  <TabList
-                    value={currentTab}
-                    centered
-                    onChange={(_, v) => setCurrentTab(v)}
-                  >
-                    <Tab value={Tabs.Batch} label={Tabs.Batch} />
-                    <Tab value={Tabs.Pricing} label={Tabs.Pricing} />
-                  </TabList>
-                </Box>
-                <Box flex={1} justifyContent="flex-end" display="flex">
-                  <ButtonWithIcon
-                    color="primary"
-                    variant="outlined"
-                    onClick={addLine}
-                    label={t('label.add-batch', { ns: 'inventory' })}
-                    Icon={<PlusCircleIcon />}
-                  />
-                </Box>
-              </Box>
+      {(() => {
+        if (isLoading) {
+          return (
+            <Box sx={{ height: isMediumScreen ? 350 : 450 }}>
+              <BasicSpinner messageKey="saving" />
+            </Box>
+          );
+        }
 
-              <TableContainer>
-                <StocktakeLinePanel
-                  batches={draftLines}
-                  update={update}
-                  value={Tabs.Batch}
-                >
-                  <BatchTable batches={draftLines} update={update} />
-                </StocktakeLinePanel>
+        return (
+          <>
+            <StocktakeLineEditForm
+              item={currentItem}
+              onChangeItem={setCurrentItem}
+              mode={mode}
+            />
+            {!currentItem ? (
+              <Box sx={{ height: isMediumScreen ? 400 : 500 }} />
+            ) : null}
+            {!!currentItem ? (
```

Comment:
I'm curious - why the `!currentItem` and `!!currentItem`?

Reply:
Um.. I'm not sure 😁